Add Livebox Router

cyr-ius opened the pull request:
## Description:

Add Orange Livebox (router)
Sensors for bandwith ans status connection
Device Trackers

**Pull request with documentation for [home-assistant.io](https://github.com/home-assistant/home-assistant.io) (if applicable):** home-assistant/home-assistant.io#10260

## Example entry for `configuration.yaml` (if applicable):
```yaml:
      livebox:
         host:
         port:
         username:
         password:
```

## Checklist:
  - [X] The code change is tested and works locally.
  - [X] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [X] There is no commented out code in this PR.
  - [X] I have followed the [development checklist][dev-checklist]

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [home-assistant.io](https://github.com/home-assistant/home-assistant.io)

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html

cyr-ius committed:
parent a91b0058229974aed578fb20e833df4454d007f9
author Cyr-ius <cyr-ius@ipocus.net> 1568662328 +0200
committer cyr-ius <cyr-ius@ipocus.net> 1578052288 +0100

Add Livebox Router

Add Wired devices option for device tracker

Collect bandwith and wan status

Add service for reboot and add switch for enable wifi

Cyr-ius commented:
Sorry Martin , big bug between the chair and the keyboard.
I took all your remarks and I corrected everything

MartinHjelmare commented:
Many of my earlier comments have not been addressed.

Cyr-ius commented:
Ok I added all the recommendations of the previous PR [26318](https://github.com/home-assistant/home-assistant/pull/26318)

Cyr-ius commented:
> Many of my earlier comments have not been addressed.

This code 
tasks = []
for component in COMPONENTS:
    tasks.append(
        hass.config_entries.async_forward_entry_unload(config_entry, component))

return all(await asyncio.gather(*tasks)) 

is Not OK . Infinite Loop or Max depth exceed.

MartinHjelmare commented on homeassistant/components/livebox/device_tracker.py:
```diff
@@ -0,0 +1,115 @@
+        await _async_setup_entities(hass, config_entry, async_add_entities, [device])
+
+    unsub = async_dispatcher_connect(
+        hass, LIVEBOX_DISCOVERY_NEW.format(DOMAIN), async_discover
```
Where do we send the signal to run the connected function?

MartinHjelmare commented on homeassistant/components/livebox/config_flow.py:
```diff
@@ -0,0 +1,129 @@
+    async def _entry_from_box(self, id):
+        """Return a config entry from an initialized box."""
+
+        # Remove all other entries of hubs with same ID or host
```
Why should we remove entries? Isn't it enough to check that we don't add the same entries?

MartinHjelmare commented on homeassistant/components/livebox/config_flow.py:
```diff
@@ -0,0 +1,129 @@
+                        "password": self.password,
+                    },
+                )
+        except Exception:
```
Please except specific exceptions for specific errors. If we know that a unique id is not found, there should be a specific exception we can catch for that case.

MartinHjelmare commented on homeassistant/components/livebox/binary_sensor.py:
```diff
@@ -0,0 +1,78 @@
+        self._box_data = box_data
+        self._box_id = id
+        self._state = None
+        self._datas = None
```
```suggestion
```

MartinHjelmare commented on homeassistant/components/livebox/sensor.py:
```diff
@@ -0,0 +1,118 @@
+        self._box_data = box_data
+        self._box_id = id
+        self._state = None
+        self._datas = None
```
```suggestion
```

MartinHjelmare commented on homeassistant/components/livebox/device_tracker.py:
```diff
@@ -0,0 +1,115 @@
+    """Represent a tracked device."""
+
+    def __init__(self, id, box_data, **kwargs):
+        """Initialize the ZHA device tracker."""
```
Stale docstring.

MartinHjelmare commented on homeassistant/components/livebox/binary_sensor.py:
```diff
@@ -0,0 +1,78 @@
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Defer binary sensor setup to the shared sensor module."""
+    box_data = hass.data[DOMAIN][DATA_LIVEBOX]
+    id = config_entry.data["id"]
```
```suggestion
    box_id = config_entry.data["id"]
```

MartinHjelmare commented on homeassistant/components/livebox/binary_sensor.py:
```diff
@@ -0,0 +1,78 @@
+    """Defer binary sensor setup to the shared sensor module."""
+    box_data = hass.data[DOMAIN][DATA_LIVEBOX]
+    id = config_entry.data["id"]
+    async_add_entities([InfoSensor(box_data, id)], True)
```
```suggestion
    async_add_entities([InfoSensor(box_data, box_id)], True)
```

MartinHjelmare commented on homeassistant/components/livebox/binary_sensor.py:
```diff
@@ -0,0 +1,78 @@
+
+    device_class = DEVICE_CLASS_CONNECTIVITY
+
+    def __init__(self, box_data, id):
```
```suggestion
    def __init__(self, box_data, box_id):
```

MartinHjelmare commented on homeassistant/components/livebox/binary_sensor.py:
```diff
@@ -0,0 +1,78 @@
+        """Initialize the sensor."""
+
+        self._box_data = box_data
+        self._box_id = id
```
```suggestion
        self._box_id = box_id
```

MartinHjelmare commented on homeassistant/components/livebox/__init__.py:
```diff
@@ -0,0 +1,161 @@
+                    }
+                }
+            }
+        await self.async_conn()
```
We need to catch potential exceptions when calling `self.async_conn`.

MartinHjelmare commented on homeassistant/components/livebox/binary_sensor.py:
```diff
@@ -0,0 +1,78 @@
+    async def async_update(self):
+        """Fetch status from livebox."""
+
+        self._dsl = await self._box_data.async_status()
```
If `_box_data.async_status` returns an empty dict, there will be key errors in the properties. We need to handle this somehow.

MartinHjelmare commented on homeassistant/components/livebox/config_flow.py:
```diff
@@ -0,0 +1,129 @@
+
+        errors = {}
+        try:
+            self.box_id = (await self._box.async_infos())["SerialNumber"]
```
`_box.async_infos` can return an empty dict. That would lead to errors here.

MartinHjelmare commented on homeassistant/components/livebox/__init__.py:
```diff
@@ -0,0 +1,161 @@
+    hass.data[DOMAIN] = {}
+    hass.data[DOMAIN][DATA_LIVEBOX] = box_data
+
+    config = await box_data.async_infos()
```
`_box.async_infos` can return an empty dict. That would lead to errors here.

MartinHjelmare commented on homeassistant/components/livebox/sensor.py:
```diff
@@ -0,0 +1,118 @@
+    async def async_update(self):
+        """Return update entry."""
+
+        self._dsl = await self._box_data.async_dsl_status()
```
This can also return an empty dict.

MartinHjelmare commented on homeassistant/components/livebox/binary_sensor.py:
```diff
@@ -0,0 +1,78 @@
+    def is_on(self):
+        """Return true if the binary sensor is on."""
+
+        if self._dsl["WanState"] == "up":
```
```suggestion
        return self._dsl["WanState"] == "up"
```

MartinHjelmare commented on homeassistant/components/livebox/device_tracker.py:
```diff
@@ -0,0 +1,115 @@
+
+    async def _update_entity(self):
+
+        device = await self._box_data.async_devices(self._device)
```
This can return an empty dict.

MartinHjelmare commented on homeassistant/components/livebox/device_tracker.py:
```diff
@@ -0,0 +1,115 @@
+
+    box_data = hass.data[DOMAIN][DATA_LIVEBOX]
+    device_trackers = await box_data.async_devices()
+    if device_trackers is not None:
```
This can never happen. `device_trackers` could be an empty dict though.

MartinHjelmare commented on homeassistant/components/livebox/__init__.py:
```diff
@@ -0,0 +1,161 @@
+"""Orange Livebox."""
+import logging
+from datetime import timedelta
+import voluptuous as vol
```
Add a blank line between standard library and 3rd party imports.

MartinHjelmare commented on homeassistant/components/livebox/__init__.py:
```diff
@@ -0,0 +1,161 @@
+import logging
+from datetime import timedelta
+import voluptuous as vol
+
```
Remove this blank line.

MartinHjelmare reviewed (changes requested):
There are still many old comments that haven't been addressed. I've tried to find most again, and I've found some new ones.

It's not very productive for me to have to go through the code and say the same things over and over. Please don't resolve the comments until you have actually pushed a fix. If something is not clear, ask me, and I'll try to answer.

Also, please think through the return types, logic flow and what can happen in the different cases. There are many places currently where some logic flow would error if it would happen. We need to handle all the different logic paths.

MartinHjelmare commented:
> is Not OK . Infinite Loop or Max depth exceed.

Please push the code, and I can have a look.

Toukassay commented:
Don't give up ;) I'm waiting for this plugin. 
Take care.

Cyr-ius commented:
this pl

> Don't give up ;) I'm waiting for this plugin.
> Take care.

the plugin works but they do not want to take it

You can copy the directory to custom_components if you want and it will appear in the Integration menu

MartinHjelmare commented:
@Cyr-ius are you planning to continue here?

Cyr-ius commented:
yes I would like to continue but what needs to be improved?

Cyr-ius commented:
> @Cyr-ius are you planning to continue here?

yes I would like to continue but what needs to be improved?

MartinHjelmare commented:
The comments need to be addressed and the build needs to pass.

omriasta committed:
Patch rachio (#30031)

* fix binary sensor offline/online

fixed self._handle_update on line 66 to produce args, kwargs.
Updated the binary sensor to check the correct index in the tuple.

* Fixed Standby switch

Set standby switch to poll in order to get status when homeassistant starts up.
Updated the index for the switch to get the status from the tuple.

Commit:
Fix failure in transform method (#30023)

* Fix failure in transform method

* Fix formatting issue

Jc2k committed:
Set unique id on homekit_controller config entries (#30035)

* Set unique id on config entries

* Changes from review

balloob committed:
Remove stream from camera after deps (#30057)

yegle committed:
Add unit_of_measurement to various Transmission sensors (#30037)

* Add unit_of_measurement to various Transmission sensors 

Without unit_of_measurement, the history graph card will not show those sensors as line chart.

* Change Counts to Torrents.

ochlocracy committed:
Refactor Alexa capabilityResources object into class, Implement Alexa semantics object (#29917)

* Refactor capabilityResources object into class.
Implement semantics object to support open, close, raise, lower utterences.
Replace covers PercentageController with RangeController.
Add semantics for covers.
Remove PowerController for covers.
Add new display categories.
Add new items to Alexa Global Catalog.
Implement garage door voice PIN code support though Alexa app.
Fixed bug with getting property for ModeController.
Fixed bug were PercentageController AdjustPercentage would exceed 100.

* Comment fixes in Tests.

* Reorder imports.

* Added additional tests for more code coverage.

* Added and additional test for more code coverage.

* Explicitly return None for configuration() if not instance of AlexaCapabilityResource.

balloob committed:
Make Hassfest stricter pt 2 (#30068)

* Make Hassfest stricter

* Fix if-condition

* Small cleanup

bieniu committed:
Init entities as unavailable when offline (#29738)

springstan committed:
Make name of nmbs live sensor customizable via unique_id (#29662)

* Make name customizable for nmbs live sensors

* Change order of the live name

* Added unique id to live nmbs sensor

* Revert changing the name of the live sensor

scop committed:
Include all SSDP data in discovery info (#28197)

* Include all SSDP data in discovery info

* Use UPnP device description as discovery info, inject some SSDP attrs

* Clean up attribute names

* Adapt existing SSDP flows to changed attribute names

* Prefix all SSDP UPnP attribute name constants with ATTR_UPNP, tweak a bit

homeassistant committed:
[ci skip] Translation update

frenck committed:
Fix update port and api key on deconz discovery config entry u… (#30088)

* Fix update port and api key on discovery config entry update

* Remove coroutine from _update_entry

frenck committed:
Fix update port and api key on deconz discovery config entry u… (#30088)

* Fix update port and api key on discovery config entry update

* Remove coroutine from _update_entry

fuzzie360 committed:
Fix homekit handling of 0 light brightness and fan speed (#29962)

* Fix homekit handling of 0 light brightness and fan speed

* Update homekit tests for new initial brightness/speed value

fuzzie360 committed:
Fix homekit handling of 0 light brightness and fan speed (#29962)

* Fix homekit handling of 0 light brightness and fan speed

* Update homekit tests for new initial brightness/speed value

michaelarnauts committed:
Rework tado component (#29246)

* Fix imports so it works in custom_components

* Rework tado component

* Code cleanup

* Remove water_heater

* Address pylint warnings

* Remove water_heater from components

* Raise PlatformNotReady when we couldn't connect

* Revert PlatformNotReady since we are not a platform

* Add debugging information

* Add fallback setting

* Import with relative path

* Address race condition

* Cleanup

* Catch 422 Errors and log the real error

* Use async_schedule_update_ha_state to update the entities

* Forgot the True

pvizeli committed:
Bump version 0.103.3

pvizeli committed:
Fix test

pvizeli committed:
Fix v2

gppanayotov committed:
Add an open window sensor for heating zones (#30090)

ochlocracy committed:
Explicitly include Alexa interface for image_processing entities. (#30101)

springstan committed:
Bump starlingbank to 3.2 (#30098)

springstan committed:
Bump starlingbank to 3.2 (#30098)

balloob committed:
Bump ring to 0.2.5 (#30103)

balloob committed:
Bump ring to 0.2.5 (#30103)

balloob committed:
Merge pull request #30096 from home-assistant/rc

0.103.3

Jc2k committed:
Don't error on removal of an ignored homekit_controller config entry (#30083)

* Don't error on ignored entries

* Don't ever call async_remove_entry or async_unload_entry for an unignored ignore config entry

basnijholt committed:
add --show-diff-on-failure to pre-commit (#30097)

This makes the traceback on a failing CI pipeline much more useful.

scop committed:
Upgrade mypy to 0.761 (#30104)

https://mypy-lang.blogspot.com/2019/12/mypy-0760-released.html
https://github.com/python/mypy/releases/tag/v0.761

scop committed:
Helpers type hint improvements (#30106)

vzahradnik committed:
Add device class attribute to modbus sensors (#30030)

cereal2nd committed:
Upgrade python-velbus (#30110)

Jc2k committed:
Add a config entry mechanism to rediscover a discovery that was ignored (#30099)

* Mechanism to rediscover a discovery that was ignored

* Add core config entry tests for new rediscover step

* Add tests for homekit_controller implementation of async_step_rediscover

* Rename rediscover to unignore

* Comment the new ignore/unignore mechanisms

Adminiuga committed:
Implement ZHA entity classes registry (#30108)

* ZHA Entity registry.
Match a zha_device and channels to a ZHA entity.

* Refactor ZHA sensor to use registry.
* Remove sensor_types registry.
* Fix ZHA device tracker battery remaining.
* Remove should_poll/force_update attributes.
* Fix binary_sensor regression.
* isort.
* Pylint.
* Don't access protected members.
* Address comments and fix spelling.
* Make pylint happy again.

balloob committed:
Clean up mobile app webhooks (#30123)

Adminiuga committed:
Refactor ZHA electrical measurement sensor. (#30130)

Adminiuga committed:
Cleanup ZHAEntity class (#30131)

Remove `_domain` attribute since we're not using it anymore.

springstan committed:
Move imports into setup function in homekit __init__.py (#30137)

springstan committed:
Move imports into setup function in homekit __init__.py (#30137)

balloob committed:
Fix test

balloob committed:
Remove requirement from entity integration (#30113)

balloob committed:
Remove requirement from entity integration (#30113)

fabaff committed:
Upgrade zeroconf to 0.24.2 (#30140)

Adminiuga committed:
Refactor ZHA binary_sensor (#30138)

* Refactor ZHA binary_sensor.

Use ZHA entity class registry for channel specific implementations.

* Remove registries.BINARY_SENSOR_TYPES dict.

* Address PR comments.

dannerph committed:
Update Integration of Keba charging station (#30125)

* fixed parsing of current to float in service set_current

* Added optional name in the config file in order to get a better entety naming (easier to find)

* fix parsing of all parameters to service calls

* addressed code review comments + updated pypi dependency

* config name imported from cont.py + minor naming changes to be more clear about the meaning of a sensor

* removed name in config again, use product name gathered from the charging station instead

* implemented suggested changes

* changed variable naming as requested

scop committed:
More helpers type improvements (#30145)

frenck committed:
Bump sqlalchemy to 1.3.12 (#30142)

cgtobi committed:
Clean up scaffold (#30135)

Adminiuga committed:
ZHA binary_sensor cleanup. (#30149)

frenck committed:
Fix deconz SSDP updating Hassio discovery (#30153)

frenck committed:
Fix deconz SSDP updating Hassio discovery (#30153)

balloob committed:
Bumped version to 0.103.4

Kane610 committed:
Allow battery value of 0 as well as make sure to not create a battery tracker if one already exist (#30155)

Kane610 committed:
Allow battery value of 0 as well as make sure to not create a battery tracker if one already exist (#30155)

balloob committed:
Merge pull request #30154 from home-assistant/rc

0.103.4

balloob committed:
Merge remote-tracking branch 'origin/master' into merge-master

ochlocracy committed:
Add support for input_number entities in Alexa integration (#30139)

* Add support for input_number entities

* Update homeassistant/components/alexa/capabilities.py

Co-Authored-By: Paulus Schoutsen <paulus@home-assistant.io>

* Removed get methods to directly access required attributes dicts.

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

Quentame committed:
Add icon to Plex sensor (#30172)

springstan committed:
Add cast to state of Dyson Air Quality Sensor (#30100)

Quentame committed:
Add icons to Freebox sensors (#30132)

gtdiehl committed:
Upgrade envoy_reader to 0.11.0 (#30179)

P-Verbrugge committed:
Updated formatting of total_blocks value (#30170)

The number of total blocks is always a round number. There can't be .1 or .11 blocks for example. The output is now always formatted with two decimals that are always 00.

heckler committed:
New date_time_utc display option added to the time_date sensor platform (#30158)

alandtse committed:
Enable config flow for Tesla (#28744)

* build: bump teslajsonpy to 0.2.0

* Remove tests

* feat: add config flow

* feat: add async

* perf: convert unnecessary async calls to sync

* feat: add charger voltage and current sensor

* feat: add options flow

* build: bump teslajsonpy to 0.2.0

* Remove icon property

* Revert climate mode change

* Remove charger sensor

* Simplify async_setup_platform

* Update homeassistant/components/tesla/sensor.py

Co-Authored-By: Paulus Schoutsen <paulus@home-assistant.io>

* Update homeassistant/components/tesla/binary_sensor.py

Co-Authored-By: Paulus Schoutsen <paulus@home-assistant.io>

* Address requested changes

* Fix pylint error

* Address requested changes

* Update codeowners

* Fix pylint error

* Address requested changes

* Address requested change

* Remove unnecessary check for existing config entry

* Load scan_interval in async_setup_entry

* Include coverage of config_flow

* Add tests for full coverage

* Address requested test changes

* Remove unnecessary init lines

* Remove unnecessary init

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

dmulcahey committed:
Protect against bad data stored in ZHA (#30183)

homeassistant committed:
[ci skip] Translation update

fabaff committed:
Upgrade zeroconf to 0.24.3 (#30187)

Quentame committed:
Fix AdGuard Home safe search sensor name (#30171)

* Fix AdGuard Home safe search sensor name

* Fix typo

garbled1 committed:
Add onewire devices and owserver remote host support (#29948)

* Add support for HobbyBoards sensors: Hub, Humidity, Moisture, PulseCounter
Add support for an owserver running on a remote host

* Cleanup and fixes for style/etc

* Forgot to convert to f-strings.

* Update with changes requested by @MartinHjelmare

* Pylint

* Fix to None

* Reverse logic for hobbyboard test to eliminate pylint complaint

* Modified manifest to list myself as codeowner

* Move some functions down into OneWireProxy instead of the top OneWire

* Oops, missed some decode()'s

* And another!

springstan committed:
Bump importlib-metadata to 1.3.0 (#30196)

* Bump importlib-metadata to 1.3.0

* Include setup.py and package_constraints.txt

springstan committed:
Bump datapoint to 0.9.5 (#30185)

ochlocracy committed:
Implement EqualizerController in Alexa for media_player. (#30159)

homeassistant committed:
[ci skip] Translation update

BKPepe committed:
Upgrade youtube_dl to 2019.12.25 (#30203)

ndonegan committed:
Pull track position from MPD status (#28579)

* Pull track position from MPD status()

This allows the progress bar to work when using the media-control card in lovelace.

* Actually commit flake8 fix!

* Extra documentation.

Mainly to trigger CI rerun.

* Updated to use self._media_position

rhadamantys committed:
Add support for enocean window handle FA 10 00 (Hoppe) (#29968)

* Added support for Somfy RTS wireless power socket and
Somfy Temperature Sensore Thermos Wirefree io

* Added code formatting fixes for commit 5faaf9c

* added support for RollerShutterRTSComponent from Somfy

* Added support for RTS roller shutter in set_cover_position

* added support for enocean window handle FA 10 00 (Hoppe)

* removed spaces in empty lines

* removal of rawdata attribute / code style

* isort fix

* PyLint fixes

* Improvements after review suggestions: rename device class to sensor type to avoid ambiguousness

* added import for DEVICE_CLASS_POWER from const.py

* removed window handle unit

HarlemSquirrel committed:
Update Vivotek camera component (#30191)

- Bump libpyvivotek version to 0.4.0
- Add digest authentication

scop committed:
Make Huawei LTE notify service name configurable (#30208)

* Default Huawei LTE notify service name to notify.huawei_lte, make configurable

Closes https://github.com/home-assistant/home-assistant/issues/29409

* Set default notify service name for uninvoked options flow

scop committed:
Add Huawei LTE integration suspend and resume services (#30207)

Useful e.g. if accessing the router web interface from another source
such as a web browser is temporarily required.

homeassistant committed:
[ci skip] Translation update

alandtse committed:
Bump teslajsonpy to 0.2.1 (#30217)

Closes #29922

PaulAnnekov committed:
bump tuyaha 0.0.5 (#30213)

bendavid committed:
Fix handling of symlinked device descriptors in keyboard_remote and move remaining sync io to executor thread pool (#30206)

* fix handling of symlinked device decriptors

* make check for symlinked paths more efficient

* make variable names pylint compliant

* move sync io during setup and device connect/disconnect to executor thread pool

* move remaining sync io during setup to executor thread pool

* remove unnecessary lambda functions

dmulcahey committed:
Bump ZHA dependencies (#30228)

* bump dependencies
* requirement files
* bump quirks to 0.0.30

homeassistant committed:
[ci skip] Translation update

frenck committed:
Bump shodan to 1.21.1 (#30234)

Kerwood committed:
DECONZ - Added support for Aqara single switch WXKG03LM (#30240)

homeassistant committed:
[ci skip] Translation update

springstan committed:
Bump python-qbittorrent to 0.4.1 (#30239)

tchellomello committed:
Bump ring to 0.2.8 to fix Oauth issues (#30245)

scop committed:
Fix Huawei LTE error message on service call without URL and routers (#30250)

Commit:
Bump dependency for HomematicIp cloud (#30237)

* Bump dependency for HomematicIp cloud

* Update test_data

xlfe committed:
Bump libpurecool to 0.6.0 (#30247)

* bump libpurecool to 0.6.0 - fixes home-assistant/home-assistant#26367

* update manifest too

scop committed:
Huawei LTE: Fix YAML options overriding ones set from GUI (#30249)

Closes https://github.com/home-assistant/home-assistant/issues/30221

Commit:
Add HmIP-MOD_TM to HomematicIP Cloud (#30255)

scop committed:
Note Huawei LTE notify service change requires restart (#30223)

Refs https://github.com/home-assistant/home-assistant/issues/30222

cereal2nd committed:
Bump python-velbus library to 2.0.32 to fix problems with the glaspanels (#30257)

balloob committed:
Whitelist Android/iOS auth callbacks (#30082)

* Whitelist Android/iOS

* Add iOS alternate flavor URLs

* Update indieauth.py

Co-authored-by: Robbie Trencheny <me@robbiet.us>

vangorra committed:
Fixing timezone issue which caused wrong selection of data to be used. (#30011)

Adminiuga committed:
Refactor ZHA channel logging (#30259)

Add channel.id property -- id unique for this the device only.

homeassistant committed:
[ci skip] Translation update

liollury committed:
Fix ble_tracker randomly pygatt thrown error (#28671)

* fix(ble_tracker): catch randomly pygatt thrown error

* fix(ble_tracker): merge except errors

mockersf committed:
Fix creating smappee sensors when remote is not active (#30270)

scop committed:
Upgrade rflink to 0.0.50, ignore_devices now supports * and ? anywhere (#30268)

https://github.com/aequitas/python-rflink/releases/tag/0.0.50
https://github.com/aequitas/python-rflink/releases/tag/0.0.49
https://github.com/aequitas/python-rflink/releases/tag/0.0.48
https://github.com/aequitas/python-rflink/releases/tag/0.0.47

homeassistant committed:
[ci skip] Translation update

Jc2k committed:
Accept homekit_controller pairing codes both with and without dashes (#30273)

* Handle MalformedPinError from homekit_python
* Handle both formats of pin codes

Jc2k committed:
Add homekit_controller service.sensor.smoke (#30269)

fabaff committed:
Upgrade beautifulsoup4 to 4.8.2 (#30274)

fabaff committed:
Add custom validator for countries (#30280)

alandtse committed:
Change default icons for Tesla components (#30288)

andygrunwald committed:
Contributing: Add note about feature suggestions and bug tracking (#30225)

taisholland committed:
Bump ihcsdk to 2.4.0 (#30279)

* Update requirements_all.txt

update ihcsdk to version 2.4.0

* Update manifest.json

upgrade to version 2.4.0 of ihcsdk

Kane610 committed:
Handle wired bug on restart (#30276)

andrewsayre committed:
Bump pysmartthings 0.7.0 (#30302)

homeassistant committed:
[ci skip] Translation update

pnbruckner committed:
Ignore google_maps updates when last_seen goes backwards (#30178)

bieniu committed:
Add GIOS integration (#28719)

* Initial commit

* Add gios to requirements

* Add tests

* Update .coveragerc file

* Run gen_requirements_all.py

* Change DEFAULT_SCAN_INTERVAL

* Better strings

* Bump library version

* run script.hassfest

* run isort

* Add icons mapping

* Remove unnecessary f-string

* Remove unnecessary listener

* Refactoring config_flow

* Add unique_id to config entry

* Change AQI states to consts in English

* Remove unused init

* Remove unused exception

* Remove private instance attribute

* Remove overwrite state property

* Fix pylint error

* Add SCAN_INTERVAL for air_quality entity

* Add _abort_if_unique_id_configured()

balloob committed:
Store capabilities and supported features in entity registry, restore registered entities on startup (#30094)

* Store capabilities and supported features in entity registry

* Restore states at startup

* Restore non-disabled entities on HA start

* Fix test

* Pass device class from entity platform

* Clean up restored entities from state machine

* Fix Z-Wave test?

frenck committed:
Fix Withings leaking time zone change into other tests (#30320)

* Fix Withings leaking time zone change in other tests

* Fix spelling error in code doc

issacg committed:
Add local_ip component (#29973)

* Added localip component

* Split config and core logic, and migrate to sensor platform (requested by @MartinHjelmare)
Also allow overriding the sensor name via the config

* Tweak docstring

Co-Authored-By: Fabian Affolter <mail@fabian-affolter.ch>

* Initial support for config entries

* Rename localip to local_ip (1/2)

* Rename localip to local_ip (2/2)

* Add test for config_flow

* Split and rename tests

* Remove unneeded code from config_flow

* Implement configuration as config entry import.  Other misc requested changes from code review.

* Fix tests

* minor code review fixes

* remove unneeded code

Co-authored-by: Fabian Affolter <mail@fabian-affolter.ch>

pschmitt committed:
Update liveboxplay and pyteleloisirs (#30093)

sbyx committed:
Add homematic host port config for HMIP-only CCUs (#30077)

* homematic: Add host port config for HMIP-only CCUs

When adding a host (CCU) to the homematic component currently the hardcoded port 2001 is used to communicate with it.
However that port is only available on the target if the target supports HM (wireless) protocol which is not the case e.g. for
the Hass.io Homematic CCU addon when running in HMIP-only mode with the HMIP-RFUSB stick.

This allows to change the port home assistant uses to talk to the CCU in order to provide services under hte homematic homain, e.g. homematic.set_variable_value

The default value for this option is the old hardcoded value this the change should be backwards compatible with existing configurations.

* Change style of config retrieval

Commit:
Bump dependency for HomematicIP Cloud (#30319)

rhadamantys committed:
Add support for Somfy RTS power socket and Somfy io Temperature sensor (#30053)

* Added support for Somfy RTS wireless power socket and
Somfy Temperature Sensore Thermos Wirefree io

* Added code formatting fixes for commit 5faaf9c

* added support for RollerShutterRTSComponent from Somfy

* Added support for RTS roller shutter in set_cover_position

* Add support for Somfy RTS power socket and Somfy io temperature sensor

* black and isort fixes

sdwilsh committed:
Reduce solaredge logging severity (#30305)

* [solaredge] Reduce Severity of Log Line

This log error happens frequently for some sites, but it shouldn't be an error.
It is expected, per the SolarEdge Monitoring API, that some sites do not
support this information, and the expected result is that this would be
empty (see comments on #27959).

Fixes #27959

* Fix a typo

Co-authored-by: Fabian Affolter <mail@fabian-affolter.ch>

pnbruckner committed:
Remove Amcrest deprecated sensors and switches (#30308)

cwildt committed:
Add support for Velux garage doors (#30214)

* Update manifest.json

* Update cover.py

Add GarageDoor

* Update to pyvlx 0.2.12

* Sort

tuxbox committed:
Add separate command and state topics for mqtt lock (#29808)

* Update lock.py

Allow different command and state topic + different command and state values.

* Formatting updated after black run

* TC updated to reflect different state & cmd values

* Abbreviations for lock states added

* additional non-default state test

* whitespaces fixed

* black formatting run

fabaff committed:
Upgrade Sphinx to 2.3.1 (#30310)

brefra committed:
Add light support to Velbus integration (#30323)

* Add light support to Velbus integration

* Add Velbus light.py to .coveragerc

* Applied black formatting

Lars-P committed:
Fix luftdaten integration by adding a sensor for pressure at sealevel (#30317)

alandtse committed:
Simplify Tesla icon get code (#30301)

* Simplify icon get code

* Remove extraneous None

Adminiuga committed:
Make the rest of ZHA platforms to use ZHA class registry (#30261)

* Refactor ZHA component tests fixtures.

* Add tests for ZHA device discovery.

* Refactor ZHA registry MatchRule.

Allow callables as a matching criteria.
Allow sets for model & manufacturer.

* Minor ZHA class registry refactoring.

Less cluttered strict_matching registrations.

* Add entities only if there are any.

* Migrate rest of ZHA platforms to ZHA registry.

* Pylint fixes.

frenck committed:
Improve Withings tests in different time zone (#30326)

* Improve Withings tests in different time zone

* Address code review comment

* Spelling error in code doc

frenck committed:
Migrate dsmr tests from coroutine to async/await (#30333)

frenck committed:
Migrate wunderground tests from coroutine to async/await (#30330)

frenck committed:
Bump pyps4-2ndscreen to 1.0.4 (#30327)

springstan committed:
Bump alarmdecoder to 1.13.9 (#30303)

frenck committed:
Migrate python_script tests from coroutine to async/await (#30335)

frenck committed:
Migrate api tests from coroutine to async/await (#30338)

frenck committed:
Migrate alexa tests from coroutine to async/await (#30332)

frenck committed:
Migrate xiaomi_miio tests from coroutine to async/await (#30329)

frenck committed:
Migrate emulated_hue tests from coroutine to async/await (#30331)

bendavid committed:
Migrate webostv to new library and make integration async with callback state updates (#29296)

* migrate webostv to new aiopylgtv version of the library and add support
for generic commands, input/button commands, and callback state updates

* update requirements

* cleanup and bump aiopylgtv version

* update webostv unit tests

* make webostv unit tests work with python 3.7

* cleanup for code checks

* cleanup and code review

* make all client request functions coroutines

* make host required for webostv configuration

* remove generic command and button functionality plus related cleanup

* fix previous track function

* update unit tests

* fix imports for unit tests

* update unit test

* further unit test updates

* remove unnecessary setup call in unit tests

* restore previous behaviour with client key config file in hass configuration directory

frenck committed:
Migrate input_* tests from coroutine to async/await (#30336)

homeassistant committed:
[ci skip] Translation update

alandtse committed:
Remove use of bin_type in Tesla component (#30315)

* Remove use of bin_type

* Convert _unit attribute to units

* Remove unnecessary variable assignment

* Change to using util library convert

BKPepe committed:
Upgrade youtube_dl to version 2020.01.01 (#30341)

fabaff committed:
Upgrade zeroconf to 0.24.4 (#30347)

frenck committed:
Migrate startca tests from coroutine to async/await (#30354)

frenck committed:
Migrate teksavvy tests from coroutine to async/await (#30353)

frenck committed:
Migrate timer tests from coroutine to async/await (#30352)

frenck committed:
Migrate yr tests from coroutine to async/await (#30351)

JeffLIrion committed:
Fix media_player example for select_source service (#30358)

fabaff committed:
Upgrade python_opendata_transport to 0.2.1 (#30348)

alandtse committed:
Fix HVAC mode for Tesla (#30287)

* Fix HVAC mode for Tesla

* Change HVAC_MODE to HEAT_COOL

AaronDavidSchneider committed:
Add attributes to fritz device_tracker (#30350)

* add attributes to device_tracker

* fix pylint

* requested changes

* docstring updated

frenck committed:
Migrate mailbox tests from coroutine to async/await (#30361)

frenck committed:
Migrate hassio tests from coroutine to async/await (#30363)

frenck committed:
Migrate websocket_api tests from coroutine to async/await (#30364)

frenck committed:
Migrate config tests from coroutine to async/await (#30366)

frenck committed:
Migrate counter tests from coroutine to async/await (#30368)

frenck committed:
Migrate owntracks tests from coroutine to async/await (#30369)

frenck committed:
Migrate no_ip tests from coroutine to async/await (#30370)

frenck committed:
Migrate camera tests from coroutine to async/await (#30372)

frenck committed:
Migrate cast tests from coroutine to async/await (#30374)

frenck committed:
Migrate ffmpeg tests from coroutine to async/await (#30375)

frenck committed:
Migrate discovery tests from coroutine to async/await (#30376)

frenck committed:
Migrate google_* tests from coroutine to async/await (#30377)

alandtse committed:
Add charging rate sensor to Tesla (#30286)

* Add charging rate sensor to Tesla

* Remove reference to bin_type

homeassistant committed:
[ci skip] Translation update

Ernst79 committed:
Restore state for Rfxtrx devices (#30309)

* Restore state rfxtrx switch

* Restore state RFXtrx lights

* Restore state RFXtrx covers

* Restore comment

* Remove line

* Remove logging

* fix black

* Fix typo

Ernst79 committed:
Remove unnessecary rfxtrx light property def (#30397)

scop committed:
Remove unnecessary string literal concatenations (#30360)

frenck committed:
Bump pre-commit to 1.21.0 (#30406)

AaronDavidSchneider committed:
fix issuecomment-570284682 (#30405)

frenck committed:
Migrate local_file tests from coroutine to async/await (#30392)

frenck committed:
Migrate fido tests from coroutine to async/await (#30391)

frenck committed:
Migrate freedns tests from coroutine to async/await (#30390)

frenck committed:
Migrate group tests from coroutine to async/await (#30389)

frenck committed:
Migrate generic tests from coroutine to async/await (#30388)

frenck committed:
Migrate media_player tests from coroutine to async/await (#30387)

frenck committed:
Migrate frontend tests from coroutine to async/await (#30386)

springstan committed:
Bump miflora and bluepy (#30411)

michaeldavie committed:
Bump env_canada to 0.0.31 (#30409)

bendavid committed:
Add generic command functionality to denonavr (#29295)

* Add generic command functionality to denonavr

* add minimal unit tests for denonavr

* fix import order

* simplify denonavr unit test

* handle domain specific service calls with dispatcher

* update unit tests

* update unit tests

* remove unnecessary return value

* fix handling of mock instances in unit tests

bendavid committed:
Add generic command/button functionality to webostv (#30379)

* add generic command/button functionality to webostv

* update codeowners

Kane610 committed:
Axis - Improve tests (#30415)

homeassistant committed:
[ci skip] Translation update

bendavid committed:
Cleanup of state handling in webostv (#30416)

* cleanup unnecessary manipulation of state variables

* update unit test

dmulcahey committed:
Remove ZHA establish device mappings function (#30423)

* remove establish_device_mappings
* inline init

scarface-4711 committed:
Implemented media_play & media_pause / push to version 0.7.11 of denonavr (#30421)

* Implement media_play & media_pause / push to version 0.7.11 of denonavr

* fix denonavr version in requirements_test_all.txt

ochlocracy committed:
Add RTSP stream support for UVC (Unifi Video Client) integration  (#30297)

* Add SUPPORT_STREAM to supported_features.

* Implement stream_source with channel RTSP URIs.

* Add Tests for Stream Support.

* Make stream_source async.

* Removed unused import.

* Re-wrote test to remove warning, and lint error.

tetienne committed:
Fix set tilt position (#30428)

scop committed:
Fix number of times seen in debug message (#30429)

Kane610 committed:
Use config entry unique id for deCONZ (#30122)

* Use config entry unique id

* Clean up

* Backwards compatiblity note

* Fix some of Balloobs comments

* Bump dependency to v66

* Black somehow missed config flow tests...

* Move set unique ID til after possibility to update existing entry

Jc2k committed:
Rework FlowManager to use inheritance (#30133)

* Pull async_finish_flow/async_create_flow out of ConfigEntries

* Towards refactoring

* mypy fixes

* Mark Flow manager with abc.* annotations

* Flake8 fixes

* Mypy fixes

* Blacken data_entry_flow

* Blacken longer signatures caused by mypy changes

* test fixes

* Test fixes

* Fix typo

* Avoid protected member lint (W0212) in config_entries

* More protected member fixes

* Missing await

agners committed:
convert to integer if rounding precision is zero (#30226)

Convert values to integer if rounding precision is zero. With that a
value which is an integer before filtering can be configured to stay
integer when using precision = 0. This also aligns behavior of filters
to how rounding behaves in tempaltes (homeassistant/helpers/template.py,
function forgiving_round).

imduffy15 committed:
Don't allow badly formed upnp devices to kill auto discovery (#30342)

Prevent the following from occurring:

```
2019-12-31 22:38:41 ERROR (MainThread) [homeassistant.core] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 85, in async_init
    return await self._async_handle_step(flow, flow.init_step, data)
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 145, in _async_handle_step
    result: Dict = await getattr(flow, method)(user_input)
  File "/usr/src/homeassistant/homeassistant/components/deconz/config_flow.py", line 182, in async_step_ssdp
    if discovery_info[ssdp.ATTR_UPNP_MANUFACTURER_URL] != DECONZ_MANUFACTURERURL:
KeyError: 'manufacturerURL'
2019-12-31 22:38:41 ERROR (MainThread) [homeassistant.core] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 85, in async_init
    return await self._async_handle_step(flow, flow.init_step, data)
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 145, in _async_handle_step
    result: Dict = await getattr(flow, method)(user_input)
  File "/usr/src/homeassistant/homeassistant/components/deconz/config_flow.py", line 182, in async_step_ssdp
    if discovery_info[ssdp.ATTR_UPNP_MANUFACTURER_URL] != DECONZ_MANUFACTURERURL:
KeyError: 'manufacturerURL'
```

elupus committed:
Mark API key as deprecated (#30402)

Service account should be used instead

Kane610 committed:
Axis - Improve tests based on feedback (#30430)

* Use MockConfigEntry, add_to_hass and hass.config_entries.async_setup

* Mock method instead of handling paths

neffs committed:
Update HomeKit position state characteristic for covers (#27867)

* HomeKit: update position state characteristic for covers

position state is a mandatory characteristic for HK window coverings

* Test position state characteristic

sergioisidoro committed:
Handle telegram event commands with args (#30254)

* Handle telegram event commands with args

* Parse message regargless of command

* Lint

* Use multiple assignment

scop committed:
Various string cleanups (#30435)

* Remove some unnecessary string concatenations

* Replace some simple str.formats with f-strings

* Replace some string concatenations with f-strings

basnijholt committed:
Add KEF speakers integration (#28959)

* add KEF speakers platform for the integration

This will work with the KEF LS50 Wireless and KEF LSX speakers.
The development of this code happened on https://github.com/basnijholt/media_player.kef

* rename DATA_KEF -> DOMAIN

* use aiokef v0.2.0 and support LSX and new features

* sort imports

* fix @MartinHjelmare's suggestions

* remove _CONFIGURING

* change STATE_UNKNOWN to None

* use lat and long for unique_id

* bump aiokef to v0.2.2

* use config[ATTR] instead of config.get(ATTR)

* use getmac

* fix case when MAC is None

* use host as instance lifetime id

* fix requirements

Kane610 committed:
WIP deCONZ - Support Fyrtur/Kadrilj battery sensors (#30403)

* Sensor.binary is None means unsupported sensor
* Don't create ordinary sensor on ZHABattery type

Jc2k committed:
isort homekit (#30437)

cyr-ius committed:
Remove dispatcher

Jc2k committed:
Don't expose flows that aren't initialised. (#30432)

* Don't expose flows that aren't initialised.

If a flow init does not return immediately then there is a window where our
behaviour is screwy:

 * Can try to configure a flow that isn't ready
 * Can show notifications for discoveries that might yet return an abort

This moves the flow discovery events and notifications to after the flow is
initialised and hides flows that don't have a cur_step from async_progress

* Fix tradfri test

* Black.

* Lint fixes

andersonshatch committed:
Check status field for UPS online binary sensor (#30144)

Avoids issue where status was previously reported as offline when
battery was low or UPS was over voltage

Kane610 committed:
deCONZ - Improve tests based on Martins feedback in Axis integration (#30438)

tchellomello committed:
Bump ring to 0.2.9 (#30439)

Kane610 committed:
Improve tests based on Martins feedback from Axis improve tests (#30442)

Jc2k committed:
Defer homekit_controller initial poll after all entities are created (#30355)

* Make sure first poll happens after sub-platforms are loaded and entities are created.

* Function dosn't need to return anything

* Don't forward entry if already forwarded it

ochlocracy committed:
Implement supported locales for Alexa capabilities (#30285)

* Implement Alexa capability supported locales.

* Check config for supported locales.

* Removed unused import.

* Applied some potpourri to improve code smell.

* Change supported_locales to set.

Kane610 committed:
Axis - Remove manual configuration and legacy config file import (#30365)

* Remove manual configuration and legacy config file import

* Remove unused imports in tests after rebase

dcramer committed:
Add Sentry component (#30422)

balloob committed:
Extract Collection helper from Person integration (#30313)

* Add CRUD foundation

* Use collection helper in person integration

* Lint/pytest

* Add tests

* Lint

* Create notification

brefra committed:
Fix async call inside sync context (#30458)

* Fix async call inside sync context

* Apply black formatting

pnbruckner committed:
Fix counter with empty config (#30454)

elupus committed:
Allow all sources and sound modes to be selected in google assistant (#30418)

* Simplify media player source list

Google don't need a whitelisted set of modes anymore. So let's just forward any mode that we have.

* Report current mode with the setting name, not a synonym

* Refactor mode generation to support other modes

* Support sound mode as mode as well

* Adjust failing test now with sound modes

homeassistant committed:
[ci skip] Translation update

Kane610 committed:
Make Axis integration use config entry unique id (#30461)

* Make Axis integration use config entry unique id

balloob committed:
Only return remote UI url if remote UI enabled (#30453)

balloob committed:
Allow adding unique device ID to mobile app registration (#30457)

* Allow adding unique device ID to mobile app registration

* Remove unknown data

* Better fix

cyr-ius committed:
Clean code

cyr-ius committed:
Fix None object and retry device if False

frenck committed:
Bump elgato to 0.2.0 (#30485)

frenck committed:
Fix WLED light transition (#30490)

frenck committed:
Add unique ID to elgato config entries (#30486)

frenck committed:
Bump adguardhome to 0.4.0 (#30483)

frenck committed:
Add unique ID to WLED config entries (#30480)

* Add unique ID to WLED config entries

* Improve robustness of the tests

frenck committed:
Improve WLED pure white support for RGBW (#30477)

frenck committed:
Bump wled to 0.2.1 (#30474)

* Bump wled to 0.2.0

* Bump wled to 0.2.1

frenck committed:
Bump twentemilieu to 0.2.0 (#30471)

Kane610 committed:
deCONZ - Allow manual configuration to update existing configuration (#30469)

* Allow manual configuration to update existing configuration

* Harmonize tests

dmulcahey committed:
Add Zigbee group binding to ZHA (#30433)

* initial group binding work
* add group cluster binding

pnbruckner committed:
Fix timer with empty config (#30463)

balloob committed:
Mobile App add device tracker to person registering app (#30460)

pvizeli committed:
support odroid-n2 on release

pvizeli committed:
bump hassio-builder version

homeassistant committed:
[ci skip] Translation update

Kane610 committed:
Axis - Allow manual configuration to update existing configuration (#30467)

* Allow manual configuration to update existing configuration

* Harmonize tests

* Understand what Elupus means...

balloob committed:
Add person reload service (#30493)

raman325 committed:
Add unique id property for Vizio devices so they get added to entity registry (#30497)

* add unique id property for Vizio devices so they get added to entity registry

* update requirements

* small change to make code more readable

* update to use f-string for consistency

* fix property name

* set unique_id to none if serial number couldnt be found

* set unique_id to none

* ESN should be globally unique so no device type needed. pyvizio will set ESN to None if ESN is unobtainable

cyr-ius committed:
Add change IP and port default adress

cyr-ius committed:
Fix Reboot service

cyr-ius committed:
Update new version aiosysbus

cyr-ius committed:
Update livebox hassfest
